partner_delivery_zone/tests/test_partner_delivery_zone.py

```diff
@@ -97,3 +98,9 @@ def test_default_line_discount_value(self):
         res = self.partner.fields_view_get(view_id=view.id, view_type='form')
         ctx = self._get_ctx_from_view(res)
         self.assertTrue('default_delivery_zone_id' in ctx)
+
+    def test_wharehouse_three_steps(self):
+        self.warehouse.delivery_steps = 'pick_pack_ship'
+        self.order.action_confirm()
+        zone = self.order.picking_ids.mapped('delivery_zone_id')
+        self.assertEqual(len(zone), 1)
```

Comment:
This is not testing what you want: it will be false positive also if only the last picking has the zone. You have to do:  `for picking in self.order.picking_ids: self.assertEqual(picking.delivery_zone_id, ...)`